internal/network/machine_network_cidr.go

```diff
@@ -50,6 +50,9 @@ func CalculateMachineNetworkCIDR(apiVip string, ingressVip string, hosts []*mode
 	if parsedVipAddr == nil {
 		return "", errors.Errorf("Could not parse VIP ip %s", ip)
 	}
+	if !mustHaveHosts && len(hosts) == 0 {
+		return "", nil
```

Comment:
We might also need to handle the case that inventory was not received (so the host is in discovering state)